pkg/overlay/server_test.go

```diff
@@ -65,3 +65,29 @@ func TestServer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewNodes(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 4, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+	// we wait a second for all the nodes to complete bootstrapping off the satellite
+	time.Sleep(2 * time.Second)
+
+	satellite := planet.Satellites[0]
+	server := overlay.NewServer(satellite.Log.Named("overlay"), satellite.Overlay, &pb.NodeStats{}, 0, 0.5)
```

Comment:
Hmm, I think the issue here is more of that specific method should belong to the database not the endpoint nor service. This makes it somewhat more harder to test. Also I started wondering whether `newNodePercentage` should be a query argument or the overlay argument?

Should the user be able to specify the limit?

Reply:
Which method should belong to the database? NewServer? I think once there's a master database a lot of this code will be obsolete and yes `newNodePercentage` will be treated more as a query. For now I think it's okay though.

Do you think it would be better for the user to set the new node percentage instead of the overlay server config? If so, I can change it.

Reply:
I mean the listing function. It takes a lot of filtering parameters and most of them don't have to do with the overlay logic itself, but rather selecting thing from the database. But, yeah it's OK for now.

As for the user specifying. I'm not sure maybe ask in slack what other think.